pkg/api/controllers/transaction_controller.go

```diff
@@ -144,33 +144,65 @@ func (ctl *TransactionController) PostTransaction(c *gin.Context) {
 	preview := ok &&
 		(strings.ToUpper(value) == "YES" || strings.ToUpper(value) == "TRUE" || value == "1")
 
-	var txData core.TransactionData
-	if err := c.ShouldBindJSON(&txData); err != nil {
+	var payload core.PostTransaction
```

Comment:
core.PostTransaction seems to be a pure api concept.
I think the structure should be located on api package (or sub package), and not in the core.

Reply:
You're right, I moved it :)